src/future/index.js

```diff
@@ -24,17 +21,15 @@ export function FormContainer({
 }) {
   const [savedDefaultValue] = useState(defaultValue);
   const [isControlled] = useState(savedDefaultValue === undefined);
+  const normalizedValue = isControlled ? value : savedDefaultValue;
   // Keep context object pure.
-  const renderContext = useMemo(() => ({ renderTag, renderComponent }), [
-    renderTag,
-    renderComponent,
-  ]);
+  const renderContext = useMemo(
+    () => ({ renderTag, renderComponent, initialValue: normalizedValue }),
```

Comment:
Passing the `initialValue` through the `RenderContext` like this seems a bit strange to me. I think it would be better to just have this available from `ReactiveValue` as `useInitialReactiveValue`. I'm going to make some changes to `ReactiveValue` though, so this is fine as-is, and I can clean it up then.